### PR TITLE
structlog 24.4.0 ❄️

### PR DESCRIPTION
**Pull request by ViridianMelody**
structlog 24.4.0 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-5756] 
- dev_url (anaconda): https://github.com/hynek/structlog/tree/24.4.0
- conda-forge: https://github.com/conda-forge/structlog-feedstock/blob/main/recipe/meta.yaml
- pypi: https://pypi.org/project/structlog/24.4.0
- pypi inspector: https://inspector.pypi.io/project/structlog/24.4.0

### Dependency for

- AnacondaRecipes/dspy-ai-feedstock/pull/1

### Explanation of changes:

- Swap to `hatchling` build system
- Update license files.
- abs file for SF channel already in `main`

[PKG-5756]: https://anaconda.atlassian.net/browse/PKG-5756?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -18,8 +18,9 @@ requirements:
   host:
     - python
     - pip
-    - setuptools
-    - wheel
+    - hatchling
+    - hatch-fancy-pypi-readme
```
Comment (lorepirri) [inline recipe/meta.yaml]:
```suggestion
    - hatch-fancy-pypi-readme >=22.8.0
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -18,8 +18,9 @@ requirements:
...
+    - hatch-vcs
   run:
     - python
     - six
```
Comment (lorepirri) [inline recipe/meta.yaml]:
```suggestion
```

not there

File: recipe/meta.yaml [hunk 3]
```diff
@@ -31,9 +32,9 @@ test:
     - freezegun >=0.2.8
     - pretend
     - python-rapidjson  # [not s390x]
+    - pytest-asyncio
```
Comment (lorepirri) [inline recipe/meta.yaml]:
```suggestion
    - pytest-asyncio >=0.17
```

is `simplejson` replaced by `python-rapidjson`?

and `pytest >=6.0`

**Review by lorepirri — COMMENTED** [review]
```
skip: True  # [py<38]
```

https://github.com/hynek/structlog/blob/24.4.0

**Review by lorepirri — APPROVED** [review]

**Review by ifitchet — APPROVED** [review]
LGTM

**Pull request merged**